lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+    on the date of the encounter.
+    """
+    # A birthdate of None will return a falsy relativedelta() object
+    delta = relativedelta(encounter_date, birthdate)
+
+    if not delta:
+        age = None
+    else:
+        # Age Ceiling
+        age = age_ceiling(delta.years)
+
+    record['age'] = age
+
+    integer_questions = [
+        'age'
+    ]
+
+    question_categories = {
+        'valueInteger': integer_questions
+    }
+
+    return create_questionnaire_response(
+        record = record,
+        question_categories = question_categories,
+        patient_reference = patient_reference,
+        encounter_reference = encounter_reference,
+        system_identifier = INTERNAL_SYSTEM)
+
+
+def get_date_from_repeat_instance(instance_id: int) -> str:
+    """
+    Returns the date associated with a REDCap repeat instance.
+    This is safe to do only when the Daily Attestation exists
+    because the Musher sets the repeat instance based on the date.
+    Returns the date as a string because that is how it is used.
+    """
+    return (STUDY_START_DATE + relativedelta(days=(instance_id -1))).strftime('%Y-%m-%d')
+
+
+def airs_map_sex(airs_sex: str) -> str:
+    sex_map = {
+        "1": "male",
+        "2": "female",
+        "3": "other",
+    }
+
+    if airs_sex in sex_map:
+        return sex_map[airs_sex]
+    else:
+        raise UnknownSexError(airs_sex)
+
+
+def airs_build_patient(enrollment: REDCapRecord) -> tuple:
+    """
+    Build a patient record using demographics. The data we need is scattered
+    across several different instruments, so they'll all need to be completed
+    in order to do this correctly.
+    """
+    return create_patient_using_demographics(
+        sex = airs_map_sex(enrollment['enr_sex']),
+        preferred_language = LANGUAGE_CODE[enrollment.project.id],
+        first_name = enrollment['scr_first_name'],
+        last_name = enrollment['scr_last_name'],
+        birth_date = enrollment['enr_dob'],
+        # TODO: verify using 'enr_mail_zip' is appropriate here;
+        # there's also 'scr_zip' which isn't necessarily the same
+        zipcode = enrollment['enr_mail_zip'],
+        record = enrollment,
+        system_identifier = INTERNAL_SYSTEM)
+
+
+def airs_build_location_info(db: DatabaseSession, cache: TTLCache,
+                             enrollment: dict) -> list:
+    """
+    Build location info from AIRS. The forms use multiple (conditional)
+    questions to check this, and different codes than what redcap.py's
+    `build_residential_location_resources()` expects, so we've got to
+    map them to one of the allowed values.
+    """
+    # Is this group housing? If so, map it to a value that results in
+    #  build_residential_location_resources() assigning it a 'lodging'
+    #  housing_type. 
+    # TODO: verify with study this is appropriate
+    if enrollment.get('enr_living_area') == '1':
+        # Maps to 'lodging'
+        housing = 'shelter'
```

Comment:
`enr_living_group` might be useful in making this data more specific.

Reply:
Unfortunately, the `enr_living_group` values don't map very well between AIRS and redcap.py. What I'm doing here is to short-circuit the check in redcap.py's [build_residential_location_resources()](https://github.com/seattleflu/id3c-customizations/blob/8b1763133a9ffa966ce3103cd58d2a4bd748939a/lib/seattleflu/id3c/cli/command/etl/redcap.py#L136) because that's only checking to see if the residence type is in a list of values, and if so, flagging it as a group home in the location. It's effectively boolean, so I picked a value that would cause the flag to be set if a participant marks that they live in a group home setting.